watertap/tools/oli_api/flash.py

```diff
@@ -0,0 +1,436 @@
+        :param survey: dictionary containing names and input values to modify
+        :param file_name: string indicating desired write location, if any
+
+        :return result: dictionary containing IDs and output streams for each flash calculation
+        """
+
+        if survey is None:
+            result = {0: oliapi_instance.call(flash_method, dbs_file_id, initial_input)}
+
+        else:
+            clones = self.modify_inputs(
+                initial_flash_input=initial_input,
+                survey=survey,
+                flash_method=flash_method,
+            )
+            result = {
+                k: oliapi_instance.call(flash_method, dbs_file_id, v)
+                for k, v in clones.items()
+            }
+
+        if file_name:
+            self.write_output_to_yaml(result, file_name)
+
+        return result
+
+    def modify_inputs(self, initial_flash_input, survey, flash_method):
+        """
+        Iterates over a survey to create modified clones of an initial flash analysis output.
+
+        :param initial_flash_input: flash analysis input to copy
+        :param survey: dictionary containing modifications for each test
+        :param flash_method: string name of flash method to use
+
+        :return clones: dictionary containing modified state variables and survey index
+        """
+
+        clones = {}
+        for i in survey.keys():
+            modified_clone = deepcopy(initial_flash_input)
+            if flash_method == "wateranalysis":
+                for param in modified_clone["params"]["waterAnalysisInputs"]:
+                    key = param["name"]
+                    if key in survey.keys():
+                        param.update({"value": survey[i][key]})
+            elif flash_method == "isothermal":
+                for param in survey:
+                    path == None
+                    if param in ["Temperature", "Pressure"]:
+                        path = modified_clone["params"][param.lower()]["value"]
+                    else:
+                        if param in modified_clone["inflows"]["values"]:
+                            path = modified_clone["inflows"]["values"][param]
+                    if path:
+                        path = survey[i][param]
+
+            else:
+                raise IOError(
+                    " Only 'wateranalysis' and 'isothermal' currently supported."
+                )
+            clones[i] = modified_clone
+        return clones
+
+    def write_output_to_yaml(self, flash_output, file_name, logging=True):
+        """
+        Writes OLI API flash output to .yaml file.
+
+        :param flash_output: dictionary output from OLI API call
+        :param filename: string name of file to write
+        :param logging: boolean switch to show printable output
+
+        :return f: string name of file written
+        """
+
+        f = f"{file_name}.yaml"
+        if logging:
+            print(f"Saving file...")
```

Comment:
We should use the `logging` module here. See, e.g., https://github.com/watertap-org/watertap/blob/dcb5c28e707980e03783db336f4efdf341c949f0/watertap/tools/parallel/multiprocessing_parallel_manager.py#L13, https://github.com/watertap-org/watertap/blob/dcb5c28e707980e03783db336f4efdf341c949f0/watertap/tools/parallel/multiprocessing_parallel_manager.py#L26, and https://github.com/watertap-org/watertap/blob/dcb5c28e707980e03783db336f4efdf341c949f0/watertap/tools/parallel/multiprocessing_parallel_manager.py#L141